rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -0,0 +1,393 @@
+                                 redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.FAST_BRIDGE_P2SH_ERP_FED)) {
+                        String message = "Tried to register a transaction with a P2SH ERP federation redeem script before RSKIP353 activation";
+                        logger.warn("[isValidPegInTx] {}", message);
+                        throw new ScriptException(message);
+                    }
+                    Script inputStandardRedeemScript = redeemScriptParser.extractStandardRedeemScript();
+                    if (activeFederations.stream().anyMatch(federation -> federation.getStandardRedeemScript().equals(inputStandardRedeemScript))) {
+                        return false;
+                    }
+
+                    Script outputScript = ScriptBuilder.createP2SHOutputScript(inputStandardRedeemScript);
+                    if (outputScript.equals(retiredFederationP2SHScript)) {
+                        return false;
+                    }
+                } catch (ScriptException e) {
+                    // There is no redeem script, could be a peg-in from a P2PKH address
+                }
+            }
+        }
+
+        Wallet federationsWallet = new BridgeBtcWallet(btcContext, activeFederations);
+        Coin valueSentToMe = tx.getValueSentToMe(federationsWallet);
+
+        boolean isUTXOsOrTxAmountBelowMinimum =
+            activations.isActive(RSKIP293) ? isAnyUTXOAmountBelowMinimum(
+                minimumPegInTxValue,
+                tx,
+                federationsWallet
+            ) : valueSentToMe.isLessThan(minimumPegInTxValue); // Legacy minimum validation against the total amount
+
+        if (!isUTXOsOrTxAmountBelowMinimum) {
+            return true;
+        }
+
+        logger.warn(
+            activations.isActive(RSKIP293)?
+                "[btctx:{}] Someone sent to the federation UTXOs amount less than {} satoshis":
+                "[btctx:{}] Someone sent to the federation less than {} satoshis",
+            tx.getHash(),
+            minimumPegInTxValue
+        );
+        return false;
+    }
+
+    @Deprecated
+    protected static boolean isMigrationTx(
+        BtcTransaction btcTx,
+        Federation activeFederation,
+        Federation retiringFederation,
+        Script retiredFederationP2SHScript,
+        Context btcContext,
+        Coin minimumPeginTxValue,
+        ActivationConfig.ForBlock activations) {
+
+        if (retiredFederationP2SHScript == null && retiringFederation == null) {
+            return false;
+        }
+
+        List<Script> standardP2shScripts = new ArrayList<>();
+        if (retiredFederationP2SHScript != null){
+            standardP2shScripts.add(retiredFederationP2SHScript);
+        }
+        if (retiringFederation != null){
+            standardP2shScripts.add(retiringFederation.getStandardP2SHScript());
+        }
```

Comment:
Why Check If They Are Not Null Again? 
We Wont Get To This Part Of The Code If Its Null Above

Reply:
retiredFederation != retiringFederation